tutorials/pages/shimmerevm-hardhat.md

```diff
@@ -0,0 +1,194 @@
+
+        unlockTime = _unlockTime;
+        owner = payable(msg.sender);
+    }
+
+    function withdraw() public {
+        // Uncomment this line, and the import of "hardhat/console.sol", to print a log in your terminal
+        // console.log("Unlock time is %o and block timestamp is %o", unlockTime, block.timestamp);
+
+        require(block.timestamp >= unlockTime, "You can't withdraw yet");
+        require(msg.sender == owner, "You aren't the owner");
+
+        emit Withdrawal(address(this).balance, block.timestamp);
+
+        owner.transfer(address(this).balance);
+    }
+}
+```
+The above is an auto-generated smart contract by hardhat. You can replace it with your own smart contract. But if you need to understand what the above smart contract does, it's a Lock contract, which will lock funds for a specific time period and allows withdrawls only past that point.
+Some important things to note are data types. `event` means that we can subscribe to it like a webhook and wait for a ping to the client side whenever Withdrawl is triggered with `emit` from inside the function `withdraw()`.
+
+### Setting up the contract
+
+- Go to `hardhat.config.js`
+- Update the `hardhat-config` with shimmer network details as shown below:
+
+```js
+module.exports = {
+  solidity: "0.8.18",
+  defaultNetwork: "shimmerevm-testnet",
+  networks: {
+    "shimmerevm-testnet": {
+      url: 'https://json-rpc.evm.testnet.shimmer.network',
+      chainId: 1071,
+      accounts: [priv_key]
+    }
+  }
+};
+```
+
+- Create `.env` file in the root to store your network id and jsonrpc in-case you'd like to keep it private. You can then refer them by `process.env.SHIMMEREVM_TESTNET_JSONRPC`.
+
+:::note
+
+Note that the file above requires DOTENV, for managing environment variables and also ethers and etherscan. Make sure to install all those packages. 
+
+Find more instructions on how to use DOTENV on [<ins>this page</ins>](https://www.npmjs.com/package/dotenv).
+
+:::
+
+### Compiling the contract
+
+To compile the contract, you first need to install Hardhat Toolbox:
+
+```sh
+npm install --save-dev @nomicfoundation/hardhat-toolbox
+```
+
+Then, simply run to compile:
+
+```sh
+npx hardhat compile
+```
+
+### Testing the Contract
+
+To run tests with Hardhat, you just need to type the following:
+
+```sh
+npx hardhat test
+```
+
+And this is an expected output:
+
+![Hardhat Test](../static/shimmer-hardhat-test.png 'Hardhat Test')
+
+### Deploying on ShimmerEVM
+
+Run this command in root of the project directory:
+
+```sh
+npx hardhat run scripts/deploy.js --network shimmerevm-testnet
+```
+
+The contract will be deployed on ShimmerEVM Testnet, and you can check the deployment status here on the [explorer](https://explorer.evm.testnet.shimmer.network/).
+
+If you want to further verify your contract, add the following to your `hardhat.config.js`:
+```js
+etherscan: {
+  customChains: [
+      {
+          network: "shimmerevm-testnet",
+          chainId: 1071,
```

Comment:
You need to add a API key dummy value here or it will complain

Reply:
Added.